src/guide/component-attrs.md

```diff
@@ -0,0 +1,106 @@
+# Non-Prop Attributes
+
+> This page assumes you've already read the [Components Basics](component-basics.md). Read that first if you are new to components.
+
+A component non-prop attribute is an attribute or event listener that is passed to a component, but does not have a corresponding property defined in [props](component-props) or [emits](component-custom-events.html#defining-custom-events). Common examples of this include `class`, `style`, and `id` attributes.
+
+## Attribute Inheritance
+
+When a component returns a single root node, non-prop attributes will automatically be added to the root node's attributes. For example, in the instance of a date-picker component:
+
+```js
+app.component('date-picker', {
+  template: `
+    <div class="date-picker">
+      <input type="datetime" />
+    </div>
+  `
+})
+```
+
+In the event we need to define the status of the date-picker component via a `data-status` property, it will be applied to the root node (i.e., `div.date-picker`).
+
+```html
+<!-- Date-picker component with a non-prop attribute -->
+<date-picker data-status="activated"></date-picker>
+
+<!-- Rendered date-picker component -->
+<div class="date-picker" data-status="activated">
+  <input type="datetime" />
+</div>
+```
+
+Same rule applies to the event listeners:
+
+```html
+<date-picker @change="submitChange"></date-picker>
+```
+
+```js
+app.component('date-picker', {
+  created() {
+    console.log(this.$attrs) // { onChange: () => {}  }
+  }
+})
+```
```

Comment:
This might benefit from an example that people can use

Reply:
Added! 👍🏻